sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/Command.scala

```diff
@@ -55,3 +55,9 @@ trait AnalysisOnlyCommand extends Command {
   // on the `AnalysisContext`
   def markAsAnalyzed(analysisContext: AnalysisContext): LogicalPlan
 }
+
+/**
+ * A command that is nested within another command after the analysis and does not have to be
+ * executed eagerly.
```

Comment:
what happens if we eagerly execute `ReplaceData`?

Reply:
Yes, we need to execute `ReplaceData` only if the delete with filters is not possible (which we can only check in the optimizer).